src/coreclr/utilcode/allocmemtracker.cpp

```diff
@@ -0,0 +1,173 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "stdafx.h"                     // Precompiled header key.
+#include "loaderheap.h"
+#include "ex.h"
+#include "pedecoder.h"
+#define DONOT_DEFINE_ETW_CALLBACK
+#include "eventtracebase.h"
+
+#ifndef DACCESS_COMPILE
```

Comment:
Move the file to (non-DAC) `UTILCODE_SOURCES` in CMakeLists.txt instead of end-to-end ifdef